locales/index.js

```diff
@@ -1,11 +1,37 @@
+import Polyglot from 'node-polyglot'
+
 import {en} from './en'
+import {es} from './es'
 import {en as enMobile} from './mobileLocales/en'
+import {es as esMobile} from './mobileLocales/es'
+import { isDesktop } from '../src/components/utils'
+
+const defaultLocale = 'en'
 
 // Language tags should follow the IETF's BCP 47 guidelines, link below:
 //https://www.w3.org/International/questions/qa-lang-2or3
 // Generally it should be a two or three charaters tag (language) followed by a two/three characters subtag (region), if needed.
-export const locales = {en}
+const locales = {en, es}
+
+const mobileLocales = {
+  en: enMobile,
+  es: esMobile
+}
+
+const unsupportedLocale = () => {
+  console.warn('Locale not supported')
+  return defaultLocale
+}
+
+const setLocale = (language) => {
+  if (!language) return defaultLocale
+  return locales[language] ? language : unsupportedLocale()
+}
 
-export const mobileLocales = {
-  en: enMobile
+export const setI18n = (language) => {
+  const locale = setLocale(language)
```

Comment:
this is still confusing. `locale` and `language` seem to be used interchangeably.

I would try to:

- Use the type in the variable name e.g. `localeString`, `phraseMap`
- Be consistent in names when the same object is used in different functions
- Make sure function names reflect what the function does. These functions no longer `set` anything.